packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -954,6 +1055,24 @@ export const createAzureCommunicationCallAdapterFromClient: (
   return new AzureCommunicationCallAdapter(callClient, locator, callAgent, deviceManager);
 };
 
+/* @conditional-compile-remove(teams-identity-support) */
+/**
+ * Create a {@link TeamsCallAdapter} using the provided {@link StatefulCallClient}.
+ *
+ * Useful if you want to keep a reference to {@link StatefulCallClient}.
+ * Consider using {@link createAzureCommunicationCallAdapter} for a simpler API.
+ *
+ * @beta
+ */
+export const createAzureCommunicationTeamsCallAdapterFromClient = async (
+  callClient: StatefulCallClient,
+  callAgent: TeamsCallAgent,
+  locator: CallAdapterLocator
+): Promise<TeamsCallAdapter> => {
+  const deviceManager = (await callClient.getDeviceManager()) as StatefulDeviceManager;
+  return new AzureCommunicationCallAdapter(callClient, locator, callAgent, deviceManager);
+};
+
```

Comment:
this file is now over 1000 lines, can we break up logic in this file into more isolated blocks in their own files

Reply:
Nice catch, @prprabhu-ms and me also discussed this yesterday and we decided to open another work item https://skype.visualstudio.com/SPOOL/_workitems/edit/3067003 for refactoring, because:
1. There is no impact on current api design
2. The refactor would be fairly large, which might involve more merge conflict for this PR (and this PR is already big enough)